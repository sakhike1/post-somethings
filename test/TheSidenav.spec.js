import { mount, createLocalVue } from "@vue/test-utils";
import TheSidenav from "@/components/Navigation/TheSidenav.vue";

// Create a localVue instance
const localVue = createLocalVue();

describe("TheSidenav", () => {
  it('emits "close" event when backdrop is clicked', async () => {
    // Mount the component with localVue
    const wrapper = mount(TheSidenav, {
      localVue,
      propsData: {
        show: true,
      },
      stubs: {
        NuxtLink: true, // Stub the NuxtLink component
      },
    });

    // Click the backdrop
    await wrapper.find(".sidenav-backdrop").trigger("click");

    // Assert that the "close" event has been emitted by the component
    expect(wrapper.emitted().close).toBeTruthy();
  });

  it('emits "close" event when nav-list is clicked', async () => {
    // Mount the component with localVue
    const wrapper = mount(TheSidenav, {
      localVue,
      propsData: {
        show: true,
      },
      stubs: {
        NuxtLink: true, // Stub the NuxtLink component
      },
    });

    // Click the nav-list
    await wrapper.find(".nav-list").trigger("click");

    // Assert that the "close" event has been emitted by the component
    expect(wrapper.emitted().close).toBeTruthy();
    expect().toMatchSnapshot();
  });
});
