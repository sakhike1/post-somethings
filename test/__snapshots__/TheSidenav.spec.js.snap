// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheSidenav emits "close" event when nav-list is clicked 1`] = `undefined`;
